src/physics/cam/co2_cycle.F90

```diff
@@ -30,30 +30,39 @@ module co2_cycle
    public co2_time_interp_fuel          ! time interpolate co2 flux
    public co2_cycle_set_ptend           ! set tendency from aircraft emissions
 
-   ! Public data
-   public data_flux_ocn                 ! data read in for co2 flux from ocn
-   public data_flux_fuel                ! data read in for co2 flux from fuel
-
-   type(co2_data_flux_type) :: data_flux_ocn
-   type(co2_data_flux_type) :: data_flux_fuel
+   ! Module data
+   type(co2_data_flux_type), public, protected :: data_flux_ocn  ! data read in for co2 flux from ocn
+   type(co2_data_flux_type), public, protected :: data_flux_fuel ! data read in for co2 flux from fuel
 
-   public c_i                           ! global index for new constituents
-   public co2_readFlux_ocn              ! read ocn co2 flux from data file
-   public co2_readFlux_fuel             ! read fuel co2 flux from data file
 
    ! Namelist variables
-   logical :: co2_flag              = .false.         ! true => turn on co2 code, namelist variable
-   logical :: co2_readFlux_ocn      = .false.         ! true => read ocn      co2 flux from date file, namelist variable
-   logical :: co2_readFlux_fuel     = .false.         ! true => read fuel     co2 flux from date file, namelist variable
-   logical :: co2_readFlux_aircraft = .false.         ! true => read aircraft co2 flux from date file, namelist variable
-   character(len=cl) :: co2flux_ocn_file  = 'unset' ! co2 flux from ocn
-   character(len=cl) :: co2flux_fuel_file = 'unset' ! co2 flux from fossil fuel
+   logical                    :: co2_flag              = .false. ! true => turn on co2 code, namelist variable
+   logical, public, protected :: co2_readFlux_ocn      = .false. ! true => read ocn      co2 flux from date file, namelist variable
+   logical, public, protected :: co2_readFlux_fuel     = .false. ! true => read fuel     co2 flux from date file, namelist variable
+   logical                    :: co2_readFlux_aircraft = .false. ! true => read aircraft co2 flux from date file, namelist variable
+
+   character(len=cl)  :: co2flux_ocn_file = 'unset'       ! co2 flux from ocn
+   character(len=cl)  :: co2flux_ocn_mesh = 'unset'       ! ESMF mesh corresponding to co2flux_ocn_file
+   integer            :: co2flux_ocn_year_first = -999    ! first year in stream to use
+   integer            :: co2flux_ocn_year_last  = -999    ! last year in stream to use
+   integer            :: co2flux_ocn_year_align = -999    ! align stream_year_first
+   character(len=cs)  :: co2flux_ocn_tintalgo = 'linear'  ! time interpolation [lower, upper, nearest, linear or coszen]
+   character(len=cs)  :: co2flux_ocn_taxmode = 'extend'   ! time extraploation [cycle, extend or limit]
```

Comment:
I think that these should also be initialized to invalid values. This would have caught the missing namelist variables.
```suggestion
   character(len=cs)  :: co2flux_ocn_tintalgo = 'NOTSET'  ! time interpolation [lower, upper, nearest, linear or coszen]
   character(len=cs)  :: co2flux_ocn_taxmode = 'NOTSET'   ! time extraploation [cycle, extend or limit]
```